var/spack/repos/builtin/packages/libtiff/package.py

```diff
@@ -3,10 +3,34 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from spack.build_systems.autotools import AutotoolsBuilder
+from spack.build_systems.cmake import CMakeBuilder
 from spack.package import *
 
+VARIANTS = [
+    # Internal codecs
+    "ccitt",
+    "packbits",
+    "lzw",
+    "thunder",
+    "next",
+    "logluv",
+    # External codecs
+    "zlib",
+    "libdeflate",
+    "pixarlog",
+    "jpeg",
+    "old-jpeg",
+    "jpeg12",
+    "jbig",
+    "lerc",
+    "lzma",
+    "zstd",
+    "webp",
+]
 
-class Libtiff(AutotoolsPackage):
+
+class Libtiff(CMakePackage, AutotoolsPackage):
```

Comment:
A tiny caveat:
```suggestion
class Libtiff(AutotoolsPackage, CMakePackage):
```
This would matter only for packages that have been installed _before_ #30738 and get reused after #30738. If CMake is first in the list they'll be guessed as CMake packages, inverting the two as Autotools packages.

Reply:
What would break without this change?

Reply:
In this specific case I think nothing, so I can go ahead merging this. In general if a build system has a `setup_dependent_build_environment` or `setup_build_dependent_package` they will not be taken into account properly in that specific reuse case (which is transitory and will "converge to non-existence" with time).

Reply:
Good to know. I'll try to watch out for that in the future. I think we can merge this as is.

Reply:
Also, to be clear, for every spec with a `build_system` variant (i.e. every spec installed after #30738) the change above would have no effect.